SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialRPCService.cpp

```diff
@@ -2,33 +2,47 @@
 
 #include "Interop/SpatialRPCService.h"
 
+#include "WorkerSDK/improbable/c_trace.h"
 #include "Interop/SpatialStaticComponentView.h"
 #include "Schema/ClientEndpoint.h"
 #include "Schema/MulticastRPCs.h"
 #include "Schema/ServerEndpoint.h"
 #include "Utils/SpatialLatencyTracer.h"
+#include "EngineClasses/SpatialNetDriver.h" // For EventTracing
+#include "EngineClasses/SpatialPackageMapClient.h" // For EventTracing
 
 DEFINE_LOG_CATEGORY(LogSpatialRPCService);
 
 namespace SpatialGDK
 {
 SpatialRPCService::SpatialRPCService(ExtractRPCDelegate ExtractRPCCallback, const USpatialStaticComponentView* View,
-									 USpatialLatencyTracer* SpatialLatencyTracer)
+									 USpatialLatencyTracer* SpatialLatencyTracer,
+									 SpatialEventTracer* EventTracer)
 	: ExtractRPCCallback(ExtractRPCCallback)
 	, View(View)
 	, SpatialLatencyTracer(SpatialLatencyTracer)
+	, EventTracer(EventTracer)
 {
 }
 
-EPushRPCResult SpatialRPCService::PushRPC(Worker_EntityId EntityId, ERPCType Type, RPCPayload Payload, bool bCreatedEntity)
+EPushRPCResult SpatialRPCService::PushRPC(Worker_EntityId EntityId, ERPCType Type, RPCPayload Payload, bool bCreatedEntity, UObject* Target, UFunction* Function)
 {
 	EntityRPCType EntityType = EntityRPCType(EntityId, Type);
 
 	EPushRPCResult Result = EPushRPCResult::Success;
 
+	if (EventTracer->IsEnabled())
+	{
+		Payload.SpanId = EventTracer->TraceEvent(FEventSendRPC(Target, Function));
+	}
+
 	if (RPCRingBufferUtils::ShouldQueueOverflowed(Type) && OverflowedRPCs.Contains(EntityType))
 	{
 		// Already has queued RPCs of this type, queue until those are pushed.
+		if (EventTracer->IsEnabled() && Payload.SpanId.IsSet())
```

Comment:
Passing around null span IDs and calling `Trace_EventTracer_ShouldSampleEvent` will perform the check you want here.